server/internal/cleaner/cleaner.go

```diff
@@ -0,0 +1,82 @@
+package cleaner
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/llmariner/api-usage/server/internal/store"
+)
+
+// Cleaner is the struct that deletes records outside the retention period.
+type Cleaner struct {
+	store           *store.Store
+	retentionPeriod time.Duration
+	ticker          *time.Ticker
+	logger          logr.Logger
+}
+
+// NewCleaner returns a new leaner.
+func NewCleaner(store *store.Store, retentionPeriod, interval time.Duration, logger logr.Logger) *Cleaner {
+	return &Cleaner{
+		store:           store,
+		retentionPeriod: retentionPeriod,
+		ticker:          time.NewTicker(interval),
+		logger:          logger.WithName("cleaner"),
+	}
+}
+
+// Run runs the poller for the Cleaner.
+func (r *Cleaner) Run(ctx context.Context) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	errCh := make(chan error)
+
+	go r.runDeletion(ctx, r.clearUsage, errCh)
```

Comment:
I don't think we need a goroutine here since we anyway wait in the following `select`.